### ISSUE TITLE
Improve performance

### ISSUE THREAD

**Issue by Stranger6667** [Stranger6667/hypothesis-graphql#14]
At the moment, data generation is relatively slow. On my machine (i7-8700K), generation of 100 queries for this schema takes on average 0.8 seconds:

```
type Book {
  title: String
  author: Author
}

type Author {
  name: String
  books: [Book]
}

enum Color {
  RED
  GREEN
  BLUE
}

input QueryInput {
  eq: String
  ne: String
}

input NestedQueryInput {
  code: QueryInput
}

type Model {
  int: Int,
  float: Float,
  string: String
  id: ID,
  boolean: Boolean
  color: Color
}

type Query {
      getModel(int: Int): Model
}
```

The version without AST nodes was ~3 times faster. Not a priority right now definitely, but we can investigate the cause in the future and improve performance

**Comment by Stranger6667** [comment]
Note: replacing `st.builds` with a simple function mapping gives ~10% better performance

**Comment by Stranger6667** [comment]
Closed via 3cfe1ababf4cd99cad194731c6bc93f8555e3628 as it gave ~65% improvement
